// @pulumi/aws/ssoadmin/customerManagedPolicyAttachmentsExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerManagedPolicyAttachmentsExclusive = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = aws.ssoadmin.getInstances({});
 * const examplePermissionSet = new aws.ssoadmin.PermissionSet("example", {
 *     name: "Example",
 *     instanceArn: example.then(example => example.arns?.[0]),
 * });
 * const examplePolicy = new aws.iam.Policy("example", {
 *     name: "TestPolicy",
 *     description: "My test policy",
 *     policy: JSON.stringify({
 *         Version: "2012-10-17",
 *         Statement: [{
 *             Action: ["ec2:Describe*"],
 *             Effect: "Allow",
 *             Resource: "*",
 *         }],
 *     }),
 * });
 * const exampleCustomerManagedPolicyAttachmentsExclusive = new aws.ssoadmin.CustomerManagedPolicyAttachmentsExclusive("example", {
 *     instanceArn: example.then(example => example.arns?.[0]),
 *     permissionSetArn: examplePermissionSet.arn,
 *     customerManagedPolicyReferences: [{
 *         name: examplePolicy.name,
 *         path: "/",
 *     }],
 * });
 * ```
 *
 * ### Disallow Customer Managed Policy Attachments
 *
 * To disallow all customer managed policy attachments, omit the `customerManagedPolicyReference` block.
 *
 * > Any customer managed policies attached to the permission set will be **removed**.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.ssoadmin.CustomerManagedPolicyAttachmentsExclusive("example", {
 *     instanceArn: exampleAwsSsoadminInstances.arns[0],
 *     permissionSetArn: exampleAwsSsoadminPermissionSet.arn,
 * });
 * ```
 *
 * ## Import
 *
 * ### Identity Schema
 *
 * #### Required
 *
 * * `instance_arn` (String) ARN of the SSO Instance.
 *
 * * `permission_set_arn` (String) ARN of the Permission Set.
 *
 * #### Optional
 *
 * * `region` (String) Region where this resource is managed.
 *
 * Using `pulumi import`, import SSO Admin Customer Managed Policy Attachments Exclusive using the `instance_arn` and `permission_set_arn` arguments, separated by a comma (`,`). For example:
 *
 * % pulumi import aws_ssoadmin_customer_managed_policy_attachments_exclusive.example arn:aws:sso:::instance/ssoins-1234567890abcdef,arn:aws:sso:::permissionSet/ssoins-1234567890abcdef/ps-1234567890abcdef
 */
class CustomerManagedPolicyAttachmentsExclusive extends pulumi.CustomResource {
    /**
     * Get an existing CustomerManagedPolicyAttachmentsExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomerManagedPolicyAttachmentsExclusive(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of CustomerManagedPolicyAttachmentsExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomerManagedPolicyAttachmentsExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customerManagedPolicyReferences"] = state?.customerManagedPolicyReferences;
            resourceInputs["instanceArn"] = state?.instanceArn;
            resourceInputs["permissionSetArn"] = state?.permissionSetArn;
            resourceInputs["region"] = state?.region;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.instanceArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            if (args?.permissionSetArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'permissionSetArn'");
            }
            resourceInputs["customerManagedPolicyReferences"] = args?.customerManagedPolicyReferences;
            resourceInputs["instanceArn"] = args?.instanceArn;
            resourceInputs["permissionSetArn"] = args?.permissionSetArn;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomerManagedPolicyAttachmentsExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomerManagedPolicyAttachmentsExclusive = CustomerManagedPolicyAttachmentsExclusive;
/** @internal */
CustomerManagedPolicyAttachmentsExclusive.__pulumiType = 'aws:ssoadmin/customerManagedPolicyAttachmentsExclusive:CustomerManagedPolicyAttachmentsExclusive';
