// @pulumi/aws/cloudfront/anycastIpList.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnycastIpList = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing a CloudFront Anycast IP List.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudfront.AnycastIpList("example", {
 *     name: "example-list",
 *     ipCount: 21,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import CloudFront Anycast IP List using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:cloudfront/anycastIpList:AnycastIpList example abcd-1234
 * ```
 */
class AnycastIpList extends pulumi.CustomResource {
    /**
     * Get an existing AnycastIpList resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AnycastIpList(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AnycastIpList.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnycastIpList.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["anycastIps"] = state?.anycastIps;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["ipCount"] = state?.ipCount;
            resourceInputs["name"] = state?.name;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.ipCount === undefined && !opts.urn) {
                throw new Error("Missing required property 'ipCount'");
            }
            resourceInputs["ipCount"] = args?.ipCount;
            resourceInputs["name"] = args?.name;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["anycastIps"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AnycastIpList.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AnycastIpList = AnycastIpList;
/** @internal */
AnycastIpList.__pulumiType = 'aws:cloudfront/anycastIpList:AnycastIpList';
