// @pulumi/aws/s3/bucketAbac.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BucketAbac = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages ABAC (Attribute Based Access Control) for an AWS S3 (Simple Storage) Bucket.
 * See the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/buckets-tagging-enable-abac.html) on enabling ABAC for general purpose buckets for additional information.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.s3.Bucket("example", {bucket: "bucket-name"});
 * const exampleBucketAbac = new aws.s3.BucketAbac("example", {
 *     bucket: example.bucket,
 *     abacStatus: {
 *         status: "Enabled",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * If the owner (account ID) of the source bucket differs from the account used to configure the Terraform AWS Provider, import using the `bucket` and `expected_bucket_owner` separated by a comma (`,`):
 *
 * Using `pulumi import`, import S3 (Simple Storage) Bucket ABAC using the `bucket` or `bucket` and `expected_bucket_owner` separated by a comma (`,`). For example:
 *
 * If the owner (account ID) of the source bucket is the same account used to configure the Terraform AWS Provider, import using the `bucket`:
 *
 * ```sh
 * $ pulumi import aws:s3/bucketAbac:BucketAbac example bucket-name
 * ```
 * If the owner (account ID) of the source bucket differs from the account used to configure the Terraform AWS Provider, import using the `bucket` and `expected_bucket_owner` separated by a comma (`,`):
 *
 * ```sh
 * $ pulumi import aws:s3/bucketAbac:BucketAbac example bucket-name,123456789012
 * ```
 */
class BucketAbac extends pulumi.CustomResource {
    /**
     * Get an existing BucketAbac resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BucketAbac(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of BucketAbac.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BucketAbac.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["abacStatus"] = state?.abacStatus;
            resourceInputs["bucket"] = state?.bucket;
            resourceInputs["expectedBucketOwner"] = state?.expectedBucketOwner;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.abacStatus === undefined && !opts.urn) {
                throw new Error("Missing required property 'abacStatus'");
            }
            if (args?.bucket === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucket'");
            }
            resourceInputs["abacStatus"] = args?.abacStatus;
            resourceInputs["bucket"] = args?.bucket;
            resourceInputs["expectedBucketOwner"] = args?.expectedBucketOwner;
            resourceInputs["region"] = args?.region;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BucketAbac.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BucketAbac = BucketAbac;
/** @internal */
BucketAbac.__pulumiType = 'aws:s3/bucketAbac:BucketAbac';
