// @pulumi/aws/redshift/idcApplication.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdcApplication = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates a new Amazon Redshift IDC application.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.redshift.IdcApplication("example", {
 *     iamRoleArn: exampleAwsIamRole.arn,
 *     idcDisplayName: "example",
 *     idcInstanceArn: exampleAwsSsoadminInstances.arns[0],
 *     identityNamespace: "example",
 *     redshiftIdcApplicationName: "example",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Redshift endpoint access using the `redshift_idc_application_arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:redshift/idcApplication:IdcApplication example example
 * ```
 */
class IdcApplication extends pulumi.CustomResource {
    /**
     * Get an existing IdcApplication resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IdcApplication(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of IdcApplication.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdcApplication.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applicationType"] = state?.applicationType;
            resourceInputs["authorizedTokenIssuer"] = state?.authorizedTokenIssuer;
            resourceInputs["iamRoleArn"] = state?.iamRoleArn;
            resourceInputs["idcDisplayName"] = state?.idcDisplayName;
            resourceInputs["idcInstanceArn"] = state?.idcInstanceArn;
            resourceInputs["idcManagedApplicationArn"] = state?.idcManagedApplicationArn;
            resourceInputs["identityNamespace"] = state?.identityNamespace;
            resourceInputs["redshiftIdcApplicationArn"] = state?.redshiftIdcApplicationArn;
            resourceInputs["redshiftIdcApplicationName"] = state?.redshiftIdcApplicationName;
            resourceInputs["region"] = state?.region;
            resourceInputs["serviceIntegration"] = state?.serviceIntegration;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.iamRoleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'iamRoleArn'");
            }
            if (args?.idcDisplayName === undefined && !opts.urn) {
                throw new Error("Missing required property 'idcDisplayName'");
            }
            if (args?.idcInstanceArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'idcInstanceArn'");
            }
            if (args?.redshiftIdcApplicationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'redshiftIdcApplicationName'");
            }
            resourceInputs["applicationType"] = args?.applicationType;
            resourceInputs["authorizedTokenIssuer"] = args?.authorizedTokenIssuer;
            resourceInputs["iamRoleArn"] = args?.iamRoleArn;
            resourceInputs["idcDisplayName"] = args?.idcDisplayName;
            resourceInputs["idcInstanceArn"] = args?.idcInstanceArn;
            resourceInputs["identityNamespace"] = args?.identityNamespace;
            resourceInputs["redshiftIdcApplicationName"] = args?.redshiftIdcApplicationName;
            resourceInputs["region"] = args?.region;
            resourceInputs["serviceIntegration"] = args?.serviceIntegration;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["idcManagedApplicationArn"] = undefined /*out*/;
            resourceInputs["redshiftIdcApplicationArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IdcApplication.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IdcApplication = IdcApplication;
/** @internal */
IdcApplication.__pulumiType = 'aws:redshift/idcApplication:IdcApplication';
