// @pulumi/aws/sagemaker/modelCardExportJob.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelCardExportJob = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage an Amazon SageMaker Model Card export job.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.ModelCardExportJob("example", {
 *     modelCardExportJobName: "my-model-card-export-job",
 *     modelCardName: exampleAwsSagemakerModelCard.modelCardName,
 *     outputConfig: {
 *         s3OutputPath: `s3://${test.example}/`,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import model card export jobs using the `model_card_export_job_arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/modelCardExportJob:ModelCardExportJob example arn:aws:sagemaker:us-west-2:123456789012:model-card/my-model-card/export-job/my-model-card-export-job
 * ```
 */
class ModelCardExportJob extends pulumi.CustomResource {
    /**
     * Get an existing ModelCardExportJob resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ModelCardExportJob(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ModelCardExportJob.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ModelCardExportJob.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["exportArtifacts"] = state?.exportArtifacts;
            resourceInputs["modelCardExportJobArn"] = state?.modelCardExportJobArn;
            resourceInputs["modelCardExportJobName"] = state?.modelCardExportJobName;
            resourceInputs["modelCardName"] = state?.modelCardName;
            resourceInputs["modelCardVersion"] = state?.modelCardVersion;
            resourceInputs["outputConfig"] = state?.outputConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.modelCardExportJobName === undefined && !opts.urn) {
                throw new Error("Missing required property 'modelCardExportJobName'");
            }
            if (args?.modelCardName === undefined && !opts.urn) {
                throw new Error("Missing required property 'modelCardName'");
            }
            if (args?.outputConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'outputConfig'");
            }
            resourceInputs["modelCardExportJobName"] = args?.modelCardExportJobName;
            resourceInputs["modelCardName"] = args?.modelCardName;
            resourceInputs["modelCardVersion"] = args?.modelCardVersion;
            resourceInputs["outputConfig"] = args?.outputConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["exportArtifacts"] = undefined /*out*/;
            resourceInputs["modelCardExportJobArn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ModelCardExportJob.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ModelCardExportJob = ModelCardExportJob;
/** @internal */
ModelCardExportJob.__pulumiType = 'aws:sagemaker/modelCardExportJob:ModelCardExportJob';
