// @pulumi/aws/sagemaker/modelCard.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelCard = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manage an Amazon SageMaker Model Card.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.ModelCard("example", {
 *     modelCardName: "my-model-card",
 *     modelCardStatus: "Draft",
 *     content: `{
 *   \\"business_details\\": {
 *     \\"business_problem\\": \\"Quality\\"
 *   },
 *   \\"intended_uses\\": {
 *     \\"intended_uses\\": \\"Testing\\"
 *   },
 *   \\"additional_information\\": {
 *     \\"caveats_and_recommendations\\": \\"Use this\\"
 *   }
 * }
 * `,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import model cards using the `model_card_name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/modelCard:ModelCard example my-model-card
 * ```
 */
class ModelCard extends pulumi.CustomResource {
    /**
     * Get an existing ModelCard resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ModelCard(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ModelCard.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ModelCard.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["content"] = state?.content;
            resourceInputs["modelCardArn"] = state?.modelCardArn;
            resourceInputs["modelCardName"] = state?.modelCardName;
            resourceInputs["modelCardStatus"] = state?.modelCardStatus;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityConfig"] = state?.securityConfig;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
        }
        else {
            const args = argsOrState;
            if (args?.content === undefined && !opts.urn) {
                throw new Error("Missing required property 'content'");
            }
            if (args?.modelCardName === undefined && !opts.urn) {
                throw new Error("Missing required property 'modelCardName'");
            }
            if (args?.modelCardStatus === undefined && !opts.urn) {
                throw new Error("Missing required property 'modelCardStatus'");
            }
            resourceInputs["content"] = args?.content;
            resourceInputs["modelCardName"] = args?.modelCardName;
            resourceInputs["modelCardStatus"] = args?.modelCardStatus;
            resourceInputs["region"] = args?.region;
            resourceInputs["securityConfig"] = args?.securityConfig;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["modelCardArn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ModelCard.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ModelCard = ModelCard;
/** @internal */
ModelCard.__pulumiType = 'aws:sagemaker/modelCard:ModelCard';
