// @pulumi/aws/ec2/vpcSecurityGroupRulesExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcSecurityGroupRulesExclusive = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Import
 *
 * Using `pulumi import`, import exclusive management of security group rules using the `security_group_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:ec2/vpcSecurityGroupRulesExclusive:VpcSecurityGroupRulesExclusive example sg-1234567890abcdef0
 * ```
 */
class VpcSecurityGroupRulesExclusive extends pulumi.CustomResource {
    /**
     * Get an existing VpcSecurityGroupRulesExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcSecurityGroupRulesExclusive(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of VpcSecurityGroupRulesExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcSecurityGroupRulesExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["egressRuleIds"] = state?.egressRuleIds;
            resourceInputs["ingressRuleIds"] = state?.ingressRuleIds;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroupId"] = state?.securityGroupId;
        }
        else {
            const args = argsOrState;
            if (args?.egressRuleIds === undefined && !opts.urn) {
                throw new Error("Missing required property 'egressRuleIds'");
            }
            if (args?.ingressRuleIds === undefined && !opts.urn) {
                throw new Error("Missing required property 'ingressRuleIds'");
            }
            if (args?.securityGroupId === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGroupId'");
            }
            resourceInputs["egressRuleIds"] = args?.egressRuleIds;
            resourceInputs["ingressRuleIds"] = args?.ingressRuleIds;
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroupId"] = args?.securityGroupId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcSecurityGroupRulesExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcSecurityGroupRulesExclusive = VpcSecurityGroupRulesExclusive;
/** @internal */
VpcSecurityGroupRulesExclusive.__pulumiType = 'aws:ec2/vpcSecurityGroupRulesExclusive:VpcSecurityGroupRulesExclusive';
