// @pulumi/aws/ram/resourceShareAssociationsExclusive.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceShareAssociationsExclusive = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Import
 *
 * Using `pulumi import`, import RAM Resource Share Association Exclusive using the `resource_share_arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:ram/resourceShareAssociationsExclusive:ResourceShareAssociationsExclusive example arn:aws:ram:eu-west-1:123456789012:resource-share/73da1ab9-b94a-4ba3-8eb4-45917f7f4b12
 * ```
 */
class ResourceShareAssociationsExclusive extends pulumi.CustomResource {
    /**
     * Get an existing ResourceShareAssociationsExclusive resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceShareAssociationsExclusive(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ResourceShareAssociationsExclusive.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceShareAssociationsExclusive.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["principals"] = state?.principals;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceArns"] = state?.resourceArns;
            resourceInputs["resourceShareArn"] = state?.resourceShareArn;
            resourceInputs["sources"] = state?.sources;
        }
        else {
            const args = argsOrState;
            if (args?.resourceShareArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceShareArn'");
            }
            resourceInputs["principals"] = args?.principals;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceArns"] = args?.resourceArns;
            resourceInputs["resourceShareArn"] = args?.resourceShareArn;
            resourceInputs["sources"] = args?.sources;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResourceShareAssociationsExclusive.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ResourceShareAssociationsExclusive = ResourceShareAssociationsExclusive;
/** @internal */
ResourceShareAssociationsExclusive.__pulumiType = 'aws:ram/resourceShareAssociationsExclusive:ResourceShareAssociationsExclusive';
